libraries/AP_Proximity/AP_Proximity_LightWareSF40C.cpp

```diff
@@ -222,7 +222,8 @@ void AP_Proximity_LightWareSF40C::set_forward_direction()
     // set forward direction
     char request_str[15];
     int16_t yaw_corr = frontend.get_yaw_correction(state.instance);
-    snprintf(request_str, sizeof(request_str), "#MBF,%d\r\n", (int)yaw_corr);
+    snprintf(request_str, sizeof(request_str), "#MBF,%d\r\n",
+             (yaw_corr > 0) ? MIN(yaw_corr, 999) : MAX(yaw_corr, -999));
```

Comment:
I'd rather replace with:
```
yaw_corr = constrain_int16(yaw_corr, 999, -999);
```

Otherwise we may use `yaw_corr` and don't realize we requested something else.

Another possible way is to simply increase the stack space to allow for complete int16_t representation (and fix the %d so you are actually printing int16_t and not int). If you want to minimize stack usage you could use this, copied from systemd source code:

```
#define DECIMAL_STR_MAX(type)                                           \  
        (2+(sizeof(type) <= 1 ? 3 :                                     \  
            sizeof(type) <= 2 ? 5 :                                     \  
            sizeof(type) <= 4 ? 10 :                                    \  
            sizeof(type) <= 8 ? 20 : sizeof(int[-2*(sizeof(type) > 8)])))  
```

Then this should fix the issue:
```
char request_str[7 + DECIMAL_STR_MAX(int16_t)];
```

Reply:
Wonder how I missed using `constrain_int16`... fixed!

Right-sizing the buffer (we should swipe that `#define`!) might do nasty things to whatever firmware is running on the Lightware.  We're already in uncharted territory with the largish numbers :-)

Reply:
?? The buffer size is used locally as a maximum only. We use strnprintf, so we always transmit the exact size we are using.